docs/data/joy/components/menu/AppsMenu.js

```diff
@@ -8,25 +8,29 @@ import MenuItem from '@mui/joy/MenuItem';
 import Apps from '@mui/icons-material/Apps';
 
 export default function AppsMenu() {
-  const [anchorEl, setAnchorEl] = React.useState(null);
-  const open = Boolean(anchorEl);
-  const handleClick = (event) => {
-    setAnchorEl(event.currentTarget);
-  };
+  const buttonRef = React.useRef(null);
+  const [open, setOpen] = React.useState(false);
+
   const handleClose = () => {
-    setAnchorEl(null);
+    setOpen(false);
   };
+
   return (
     <Box>
       <IconButton
+        ref={buttonRef}
         id="apps-menu-demo"
         aria-label="Applications"
-        aria-controls={open ? 'apps-menu' : undefined}
+        aria-controls={'apps-menu'}
         aria-haspopup="true"
         aria-expanded={open ? 'true' : undefined}
         variant="plain"
         color="neutral"
-        onClick={handleClick}
+        onClick={() => {
+          if (!open) {
+            setOpen(true);
+          }
+        }}
```

Comment:
Need to be this, otherwise the menu does not close when clicking on the button again.
```suggestion
        onClick={() => setOpen(!open)}
```

Reply:
x the rest of the demos.

Reply:
resloved here https://github.com/mui/material-ui/pull/36917/commits/f30ad68bf98f78b66be320b9cee2f625056a30c5